### ISSUE TITLE
Transport stops should only be for selected mode of transport

### ISSUE THREAD

**Issue by Friptick** [osmandapp/OsmAnd#20640]
### Description

In `Configure map > Transport`, the option `Transport stops` is all but useless if you are not traveling by bus. That is because in any city almost all "transport stops" are bus stops.  

### Steps to reproduce

1. In `Configure map > Transport`, enable `Transport stops` but disable `Bus routes`
![tr2](https://github.com/user-attachments/assets/5e57ebd8-59b3-4be9-aef1-f46b75a232a4)

2. Look at map. Wince.
![tr3](https://github.com/user-attachments/assets/546bc42e-c697-44bc-bbab-9ebf741726a5)

### Actual result

Whole map will be plastered with bus stops, which I explicitly do not want to see because I disabled bus routes.

### Expected result

See only metro and train stops.

### Your Environment (required)

```
OsmAnd Version: 4.3.4
```

**Comment by yuriiurshuliak** [comment]
The bug was not reproduced. You mentioned that you enabled Transport stops but disabled Bus routes, yet bus stops are still visible. Please note that the Transport stops option controls the display of all stops on the map, including bus stops. If you disable Transport stops and enable subway and train routes instead, only those will be shown.

We also recommend updating your app to the latest version, as there may be outdated issues that have already been resolved.

OsmAnd~ 4.9.0#3020m, released: 2024-08-19

<table style="width:100%">
  <tr>
<th><img src="https://github.com/user-attachments/assets/40a5090c-5d0e-417b-8bb2-333472dcc550"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/user-attachments/assets/18dab05a-f82d-448f-8cb4-2973591a26ca"  width="300" height="auto"/> </th>
  </tr>
</table>

<table style="width:100%">
  <tr>
<th><img src="https://github.com/user-attachments/assets/1c174850-7940-46b1-8404-cafe9cf0c2c4"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/user-attachments/assets/59e43ada-08d1-4619-9211-8988199b7e0f"  width="300" height="auto"/> </th>
  </tr>
</table>

**Comment by Friptick** [comment]
Yes. I understand all that.
>Please note that the Transport stops option controls the display of all stops on the map, including bus stops.

That is the bug. Sure, you'll probably say it's a feature, but it's not. It's obviously unexpected behavior.

To recap the situation:
-  The user **selects** "Transport stops"
-  The user **unselects** "Bus routes"
-  **The user sees almost nothing but bus stops** (and, de facto, they generally cover up all the other kinds of stop, making this feature useless for anything but buses)

I understand that resolving this may be hard (because of the available OSM data) but this is evidently a UX bug.

**Comment by vshcherb** [comment]
You are right transport stops are not related to a specific transport and UI doesn't suggest it should be.

**Comment by Friptick** [comment]
Yes yes. I understand that it's a "feature" but you're avoiding the issue. Again:

1. Imagine you want to see where all the metro stops are in your city. Very common scenario! So you enable `Transport stops` and disable everything but `Metro lines`
2. Look at the screenshots above

Are you really saying that an ordinary user is going to expect this behavior?

Is there anything that they can do to solve the problem? Is there anything that the app can do about it, given the OSM dataset?

**Comment by vshcherb** [comment]
Don't enable transport stops, you can stil lclick on stations and see them

**Comment by Friptick** [comment]
Yes, this is currently the only feasible option. The fundamental problem is that "transport stop" is pretty meaningless given that almost every one of them is a bus stop. But yes, I understand that this is an issue with the underlying OSM data.  

I did discover a solution which has really improved things: the recently implemented new "Pale" road style (#20208). This makes the map so much more usable for public transport in general.

**Comment by Friptick** [comment]
Just did some research. If OsmAnd confines itself to OSM's "Public Transport v2" schema, this UX bug seems quite easy to fix. To show only underground stops (i.e. when the users has said they only want to see "Underground routes"), you would just need to filter for `subway=yes`.

If OsmAnd really needs this current option for "Show literally every single public transport stop including all the forms of transport I don't use" (personally I'm skeptical), then it could be preserved in parallel.

To see the ongoing problem, just consult the various screenshots above.
